src/main/java/org/mitre/synthea/export/FhirR4.java

```diff
@@ -737,11 +742,42 @@ private static BundleEntryComponent encounter(Person person, BundleEntryComponen
     classCode.setCode(EncounterType.fromString(encounter.type).code());
     classCode.setSystem("http://terminology.hl7.org/CodeSystem/v3-ActCode");
     encounterResource.setClass_(classCode);
+    
     encounterResource
         .setPeriod(new Period()
             .setStart(new Date(encounter.start))
             .setEnd(new Date(encounter.stop)));
 
+    /* Length of Stay
+
+    Encounter class has both setPeriod and setLength. The first one (which is being implemented up here) sets the
+    start and end date of the encounter. It is easy to implement because setStart and setEnd (from the period class)
+    recieve a Date object.
+    On the other hand, setLength recieves a Duration object. 
+
+    } 
+    */
+    Duration length_of_stay = new Duration();
+    // setValue gets inhereted from Quantity class
+    
+    Date date_start = new Date(encounter.start);
+    Date date_stop = new Date(encounter.stop);
+    long diff = date_stop.getTime() - date_start.getTime();//as given
+
+    // long seconds = TimeUnit.MILLISECONDS.toSeconds(diff);
+    long minutes = TimeUnit.MILLISECONDS.toMinutes(diff);
+
+    // Set information in the Duration object
+    length_of_stay.setCode("m")
+                  .setSystem("http://unitsofmeasure.org")
+                  .setValue(minutes);
+    
+    // Once the Duration object is constructed and defined, create and extension and add this information
+    String url = "https://verily-src.github.io/vhp-hds-vvs-fhir-ig/StructureDefinition/length-of-stay";
```

Comment:
I'm not sure if this is publicly accessible if we want to merge the PR to the Sythea repo.

Reply:
Added consideration to the GDoc detailing this PR